### ISSUE TITLE
GitHub Desktop ...?

### ISSUE THREAD

**Issue by nelsonic** [nelsonic/nelsonic.github.io#440]
While reviewing a "Mega PR" (_tens of thousands of lines of code!_)
I realised that I had no way of _efficiently_ reviewing the changes 
(_I was reviewing in the browser but each time I tried to expand the file to see all the changes it would "hang" in Chrome and Firefox ... because the browser had already loaded a lot of data..._)

> _Additionally_ I was unable to "load up" several PRs when I had good internet and then view them offline (_when no internet is available..._) because again, my Web Browser would grind to a halt...!

Decided to try GitHub Desktop to see if it can help: https://desktop.github.com
![image](https://user-images.githubusercontent.com/194400/34382077-659bd4ec-eb04-11e7-99b7-c7e8d084915d.png)

Obviously it requires "Full Access": 
![image](https://user-images.githubusercontent.com/194400/34382056-4dc76aca-eb04-11e7-9fab-59e3766e6086.png)

Testing it now...
After authenticating I see this:
![image](https://user-images.githubusercontent.com/194400/34382456-efede16a-eb06-11e7-8598-0dcb009341a1.png)

Not a very helpful "home screen" for an _existing_ user ... 🙄

**Comment by nelsonic** [comment]
An example of an "XXL" _Open Source_ PR: https://github.com/dwyl/english-words/pull/35
![image](https://user-images.githubusercontent.com/194400/34382646-47fd7194-eb08-11e7-8fc9-466a85f2b108.png)

Sadly when I attempt to view the PR in GitHub desktop I get the following Error:
![github-desktop-file-too-large](https://user-images.githubusercontent.com/194400/34382699-9dea5e6e-eb08-11e7-8b32-3c3b1fba21fe.png)

In other words, for Large Pull Requests with many changes in files, GitHub Desktop is _useless_. 😞

**Comment by nelsonic** [comment]
Deleted the app. _useless_. 😕 
Might consider an alternative Git GUI but for now just going to use CLI. 🙄

**Comment by MysticMight** [comment]
@nelsonic have you tried SourceTree, it is similar to the GitHub Desktop app, though different layout etc. I am not sure if this will fix your issue though, I haven't dealt with large file commits.

I did a bit of research on the file limit and it looks like Git-LFS was designed for large files (https://git-lfs.github.com/) source: https://github.com/desktop/desktop/issues/2213.
By default it can be used with command line but it also looks like it can be used with Github Desktop: https://github.com/blog/2079-managing-large-files-with-git-lfs

Edit: This is perhaps also useful for others that come across this 'issue', though I haven't tested it because I don't deal with large file commits.

**Comment by nelsonic** [comment]
@MysticMight thanks for sharing your research. 🔍 👍 
SourceTree was the original app I/we used for source control ... 
Sadly, SourceTree does not display "Large Files" ...
also, I stopped using SourceTree because it requires **`Admin`** Access 
to ***everything*** on my GitHub account:
![image](https://user-images.githubusercontent.com/194400/36667022-63fa9672-1ae4-11e8-8327-45d99daa07f7.png)
I want to authorise an app to have access to only a **_specific_ subset** of my projects and only with the **_minimum_ privileges** required to get it's "work" done. i.e. push & pull code.
see: https://en.wikipedia.org/wiki/Principle_of_least_privilege
